### PR TITLE
$: Add rzshell façade

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr is an initial conversion of `$` to rzshell. Internally, it's still OLDINPUT code. Note that quoting rules have changed, I think for the better.

**Test plan**

All builds are green.

**Closing issues**

Tracking issues: #1583, #1342

### CODE REVIEW

File: librz/core/cmd/cmd_alias.c [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	"$", "foo", "alias for 's foo' (note that command aliases can override flag resolution)",
+	"$", "dis=base64:cGRm", "alias this base64 encoded text to be executed when $dis is called",
+	"$", "dis=$hello world", "alias this text to be printed when $dis is called",
+	"$", "dis=-", "open cfg.editor to set the new value for dis alias",
+	"\"$", "dis=af;pdf\"", "create command - analyze to show function", // Move quotation marks inside
+	"\"$", "test=#!pipe node /tmp/test.js\"", "create command - rlangpipe script", // Move quotation marks inside
+	"$", "dis=", "undefine alias",
+	"$", "dis", "execute the previously defined alias",
+	"$", "dis?", "show commands aliased by $dis",
+	"$", "dis?n", "show commands aliased by $dis, without a new line",
```
Comment (wargio) [inline librz/core/cmd/cmd_alias.c]: ~~probably these should be defined in the yaml file.~~
These are defined in tha yaml file, but they should never be called when `$?` is invoked.

Comment (kazarmy) [reply librz/core/cmd/cmd_alias.c]: Ok 3abe7b782ba95fd2a3c19d29e06d4c2f64e7e51c

**Comment by wargio** [comment]
I love how github is scared by UTF-8. `The head ref may contain hidden characters: "$-add-rzshell-fa\u00E7ade"`

File: librz/core/cmd/cmd_alias.c [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		for (i = 0; i < count; i++) {
+			char *v = rz_cmd_alias_get(core->rcmd, keys[i], 0);
+			char *q = rz_base64_encode_dyn((const ut8 *)v, strlen(v));
+			if (buf[2] == '*') {
+				rz_cons_printf("%s=%s\n", keys[i], v);
+			} else {
+				rz_cons_printf("%s=base64:%s\n", keys[i], q);
+			}
+			free(q);
+		}
```
Comment (wargio) [inline librz/core/cmd/cmd_alias.c:109]: probably we could extract each function.

File: librz/core/cmd/cmd_alias.c [hunk 3]
```diff
@@ -0,0 +1,159 @@
+// SPDX-FileCopyrightText: 2024 RizinOrg <info@rizin.re>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include "rz_cmd.h"
+#include "rz_core.h"
+
+static int rz_cmd_alias(void *data, const char *input) {
```
Comment (wargio) [inline librz/core/cmd/cmd_alias.c:8]: maybe add a comment here as `TODO` where you explain that this needs to be refactored.

**Review by wargio — APPROVED** [review]

File: librz/core/cmd/cmd_alias.c (hunk 3 above)
Comment (kazarmy) [reply librz/core/cmd/cmd_alias.c:8]: ok 707f3464e6d499a09fc307547a37029557860b9b

**Comment by kazarmy** [comment]
> I love how github is scared by UTF-8. ...

ASCII as the lowest common denominator

**Comment by arrowd** [comment]
This change regenerated `subprojects/rizin-shell-parser/src/parser.c` effectively reverting https://github.com/rizinorg/rizin/pull/4165
The new release fails to build for me :<

**Comment by kazarmy** [comment]
I think there needs to be a way to ensure that there is no repeat of #4325

**Pull request merged**